### PR TITLE
Add focus outline back to default buttons

### PR DESCRIPTION
**Pull request by yvonnetangsu**
# READY FOR REVIEW

# Summary
- Put focus outline back on default buttons and let developers remove them in their own projects if needed, since it's really difficult to put outline back once it's been removed (no good CSS code to reset it). Having the outline as default helps with accessibility.
- Also remove extra bottom margin on some of the CTA variants.

# Needed By (Date)
- Whenever

# Urgency
- N/A

# Steps to Test

1. Pull this branch and compile styleguide
2. Go to the Components -> button section and tab through the buttons, check that the browser default outline ring is back when button is on focus. 
3. Look at the Site Search component and check that the magnifying glass icon still has outline when tab to it.
4. Check Main Nav and Masthead component at MD breakpoint or smaller. See that the hamburger menu toggle button has no outline on focus (this is an exception case that has no focus outline since it has the thick border underneath already).

# Affected Projects or Products
- Decanter
- Redwood WP theme

# Associated Issues and/or People
- @JBCSU

### CODE REVIEW

File: core/src/scss/elements/input/_input.scss [hunk 1]
```diff
@@ -8,7 +8,8 @@
 input,
 textarea,
 select {
-  appearance: none;
+  -moz-appearance: none;
```
Comment (sherakama) [inline core/src/scss/elements/input/_input.scss]: We shouldn't need vendor prefixes as the autoprefixer should take care of this for us.

Comment (yvonnetangsu) [reply core/src/scss/elements/input/_input.scss]: Oh @sherakama , this is the weird one, the one without prefix is invalid, so the only forms that are actually valid are the prefix ones so I thought I'd remove the un-prefixed one.

Comment (JBCSU) [reply core/src/scss/elements/input/_input.scss]: This comment from https://developer.mozilla.org/en-US/docs/Web/CSS/appearance concerns me:

> **Compatibility note**: If you wish to use this property on websites, you should test it very carefully. Although it is supported in most modern browsers, its implementation varies widely. In older browsers, even the keyword none does not have the same effect on all form elements across different browsers, and some do not support it at all. The differences are smaller in the newest browsers.

Should we be using `appearance` at all?

File: core/src/scss/utilities/placeholders/components/_cta.scss [hunk 2]
```diff
@@ -22,6 +22,7 @@
 
   .su-cta__button {
     @include button-big;
+    @include margin(null null 0);
```
Comment (JBCSU) [inline core/src/scss/utilities/placeholders/components/_cta.scss]: Why was this necessary? Just curious.

Comment (yvonnetangsu) [reply core/src/scss/utilities/placeholders/components/_cta.scss]: In the current one, when you tab to the CTA, in a couple of them, there's a white space the whole CTA (more obvious when you tab to it so you see the focus outline).

File: core/src/scss/elements/input/_input.scss (hunk 1 above)
Comment (yvonnetangsu) [reply core/src/scss/elements/input/_input.scss]: Actually, yeah, @JBCSU , I'm not really sure we need it for the browsers that we're supporting (2 versions back). @sherakama , what do you think? Could we get rid of appearance?

Comment (sherakama) [reply core/src/scss/elements/input/_input.scss]: +1 to removing it.

File: core/src/scss/utilities/placeholders/components/_cta.scss (hunk 2 above)
Comment (JBCSU) [reply core/src/scss/utilities/placeholders/components/_cta.scss]: Ok, I see why you want to remove the bottom margin on the CTA button. But this has the effect of changing the vertical spacing of the entire CTA. Should we not just move the bottom margin to the whole CTA, vs. eliminating it?

File: core/src/scss/elements/input/_input.scss (hunk 1 above)
Comment (JBCSU) [reply core/src/scss/elements/input/_input.scss]: +1 also. Please remove it.

File: core/src/scss/utilities/placeholders/components/_cta.scss (hunk 2 above)
Comment (yvonnetangsu) [reply core/src/scss/utilities/placeholders/components/_cta.scss]: I believe we don't actually want any external spacing that's outside any Decanter elements (no spacing outside the top level wrapper) so I'm removing it all together.

File: core/src/scss/elements/input/_input.scss (hunk 1 above)
Comment (yvonnetangsu) [reply core/src/scss/elements/input/_input.scss]: Gladly 😄

File: core/src/scss/utilities/placeholders/components/_cta.scss (hunk 2 above)
Comment (JBCSU) [reply core/src/scss/utilities/placeholders/components/_cta.scss]: Hmmm. Yes, I agree with that in principle. However, the fact is we did effectively have external spacing on the CTA component. If people upgrade to the latest master, their layout will be affected if they use CTA's. Maybe that's ok, but I want to make a conscious decision about it.

@sherakama, your thoughts? Are CTA's used on any of your decanter-based sites? Is SWS okay with the bottom margin going away?

File: core/src/scss/elements/input/_input.scss (hunk 1 above)
Comment (yvonnetangsu) [reply core/src/scss/elements/input/_input.scss]: Ok, it's gone from both input and button-base

File: core/src/scss/utilities/placeholders/components/_cta.scss (hunk 2 above)
Comment (yvonnetangsu) [reply core/src/scss/utilities/placeholders/components/_cta.scss]: That's a good point about backward compatibility 😄

Comment (sherakama) [reply core/src/scss/utilities/placeholders/components/_cta.scss]: CSS can change and the styles applied to elements can change, esp, if the new code keeps the end result the same. Keep in mind, we try to avoid top margin and try to only apply margin to the bottom of elements to stay consistent.
